modules/EPMC.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+
+import argparse
+import sys
+from pyspark import *
+from pyspark.sql import *
+from pyspark.sql.types import *
+from pyspark.sql.functions import *
+from pyspark.sql.window import Window
+import logging
+
+
+# example run local, make sure you dont pass many parquet files as input to not run out of mem
+# also, double check you put the ram size properly in the local settings in the SparkConf map
+# python modules/EPMC.py --local \
+#        --cooccurrenceFile /home/mkarmona/src/opentargets/data/platform/epmc/epmc-cooccurrences \
+#        --outputFile test_out
+#        --logFile epmc_parser.log
+
+
+def main():
+
+    ##
+    ## Parsing parameters:
+    ##
+    parser = argparse.ArgumentParser(description='This script generates target/disease evidence strings from ePMC cooccurrence files.')
+    parser.add_argument('--cooccurrenceFile', help='Partioned parquet file with the ePMC cooccurrences', type=str, required=True)
+    parser.add_argument('--outputFile', help='Resulting evidence file saved as compressed JSON.', type=str, required=True)
+    parser.add_argument('--logFile', help='Destination of the logs generated by this script.', type=str, required=False)
+    parser.add_argument('--local', help='Destination of the logs generated by this script.', action='store_true', required=False, default=False)
+    args = parser.parse_args()
+
+    # extract parameters:  
+    cooccurrenceFile = args.cooccurrenceFile
+
+
+    # Initialize logger based on the provided logfile. 
+    # If no logfile is specified, logs are written to stderr 
+    logging.basicConfig(
+        level=logging.INFO,
+        format='%(asctime)s %(levelname)s %(module)s - %(funcName)s: %(message)s',
+        datefmt='%Y-%m-%d %H:%M:%S',
+    )
+    if args.logFile:
+        logging.config.fileConfig(filename=args.logFile)
+    else:
+        logging.StreamHandler(sys.stderr)
+
+    # Parse output file:
+    out_file = args.outputFile
+
+    ##
+    ## Initialize spark session
+    ##
+    global spark
+    local = True
+
```

Comment:
Global variables should generally be avoided; and in this case it's also not needed since the variable `spark` only exists within the `main()` function.

Also, the variable `local` appears to be not used.

```suggestion
```